nnvm/python/nnvm/frontend/onnx.py

```diff
@@ -186,6 +186,7 @@ def _impl_v1(cls, inputs, attr, params):
             inputs[0] = _sym.transpose(inputs[0], axes=(1, 0))
         if not transB:
             inputs[1] = _sym.transpose(inputs[1], axes=(1, 0))
+        inputs[0] = _sym.flatten(inputs[0])
```

Comment:
Can we verify if flatten is necessary? i.e. check the dimension  and only insert the flatten if necessary

Reply:
@nishi-t can you act on this comment?

Reply:
It is strange that you cannot get the output shape when the input[0] is pooling. Maybe can you look into what happens here?

Reply:
Thank you for your advice. I analyzed what happened here and found out a few things.
In the stage of `nnvm.frontend.from_onnx`, maybe infer_shape does not work completly, because the input's shape may not be decided.

When I putted the following code before the flatten and ran it, out_shapes was `[[]]`

```
g = _graph.create(inputs[0])
shape_dict = {k: v.shape for k, v in params.items()}
_, out_shapes = graph_util.infer_shape(g, **shape_dict)                           
```

In following lines called from above infer_shape, `dshape.ndim()` became 0 and return flase.

https://github.com/dmlc/tvm/blob/master/nnvm/src/top/nn/pooling.cc#L33
https://github.com/dmlc/tvm/blob/master/nnvm/src/top/nn/convolution.cc#L62

It seems difficult to use infer_shape without making any changes for my purpose. Please let me know what you think.

Reply:
@zhreshold can you comment on this?

Reply:
infer shape is not available because input shape is not ready. Currently I cannot find a way to detect if flatten is necessary as well.

Reply:
OK, maybe an easier way is to add an optimization to remove flatten in nnvm when it is not necessary. I am going to merge this in for now